neural_network/input_data.py

```diff
@@ -46,16 +44,16 @@ def _read32(bytestream):
 def _extract_images(f):
```

Comment:
Single letter variable names make the code look old school.  Would it be possible upgrade to more self-documenting variable names and add type hints?